controller/user.js

```diff
@@ -0,0 +1,237 @@
+);
+
+const schema = Joi.object({
+  email: Joi.string()
+    .email({
+      minDomainSegments: 2,
+      tlds: { allow: ["com", "net", "pl"] },
+    })
+    .required(),
+  password: Joi.string().pattern(passwordPattern).required(),
+});
+
+const SECRET = process.env.SECRET_WORD;
+
+const create = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  try {
+    const newUserTemplate = {
+      email,
+      password,
+    };
+
+    const validationByJoi = schema.validate(newUserTemplate);
+
+    if (validationByJoi.error) {
+      const validationErrorMessage = validationByJoi.error.details[0].message;
+      return res
+        .status(400)
+        .json(
+          genereteJSON(
+            "error",
+            400,
+            "error message",
+            validationErrorMessage || "Unsuccessful validation",
+          ),
+        );
+    }
+
+    // If getUserByEmail doesn't find user with argument email, it returns null
+    const isEmailUsed = Boolean(await service.getUserByEmail(email));
+
+    if (isEmailUsed) {
+      return res
+        .status(409)
+        .json(genereteJSON("error", 409, "error message", "Email in use"));
+    }
+
+    const hash = await hashPassword(password);
+
+    const newUser = { ...newUserTemplate, password: hash };
+
+    await service.addUser(newUser);
+
+    res.status(201).json(
+      genereteJSON("success", 201, "user", {
+        email,
+        subscription: "starter",
+      }),
+    );
+  } catch (error) {
+    console.log(error);
+    next();
+  }
+};
+
+const logIn = async (req, res, next) => {
+  const { email, password } = req.body;
+
+  try {
+    const validationByJoi = schema.validate({
+      email,
+      password,
+    });
+
+    if (validationByJoi.error) {
+      const validationErrorMessage = validationByJoi.error.details[0].message;
+      return res
+        .status(400)
+        .json(
+          genereteJSON(
+            "error",
+            400,
+            "error message",
+            validationErrorMessage || "Unsuccessful validation",
+          ),
+        );
+    }
+
+    const user = await service.getUserByEmail(email);
+
+    const result = await verifyPassword(password, user.password);
+
+    if (!result) {
+      return res
+        .status(401)
+        .json(
+          genereteJSON(
+            "error",
+            401,
+            "error message",
+            "Email or password is wrong",
+          ),
+        );
+    }
+
+    const payload = {
+      _id: user._id,
+    };
+
+    const token = jwt.sign(payload, SECRET, { expiresIn: 60 * 5 * 1000000 });
+
+    const responseBody = {
+      token,
+      user: { email: user.email, subscription: user.subscription },
+    };
+
+    res.json(genereteJSON("success", 200, "body", responseBody));
+  } catch (error) {
+    console.log(error);
+    next();
+  }
+};
+
+const logOut = async (req, res, next) => {
```

Comment:
Ta metoda powinna używać auth middleware i mieć dostęp do user wewnątrz req. Użyj tego i wyloguj użytkownika bez kolejnego sprawdzenia czy taka osoba jest w bazie danych. Jeśli dotarła aż tutaj, to musi być.